### PR TITLE
Add needs_rewrite_passes and content_changed methods

### PR DESCRIPTION
**Pull request by PoignardAzur**
These methods will be used with #639 to implement a fixpoint loop for rewrite passes.

This also lets us remove `mark_changed`, since we can use `content_changed` for the same purpose.

### CODE REVIEW

File: xilem/src/lib.rs [hunk 1]
```diff
@@ -273,12 +271,6 @@ impl ViewCtx {
         }
     }
 
-    pub fn mark_changed(&mut self) {
-        if cfg!(debug_assertions) {
-            self.view_tree_changed = true;
-        }
-    }
-
```
Comment (Philipp-M) [inline xilem/src/lib.rs:281]: I mean I agree with that change, but it doesn't seem related to the title of this PR?

Comment (PoignardAzur) [reply xilem/src/lib.rs:281]: The idea is Xilem now uses `needs_rewrite_passes` to check the same thing.

**Review by Philipp-M — APPROVED** [review]
LGTM.

But I wonder whether we need this (previously `mark_changed` now `content_changed`) at all? I think it's kind of redundant?

**Comment by PoignardAzur** [comment]
Eh, maybe. It'll be useful on the Masonry side. We might remove it on the Xilem side.

**Comment by jaredoconnell** [comment]
Nice. This simplifies the code and prevents accidentally leaving a widget stale.
